### PR TITLE
PR: Add new options for font size and bottom edge for inline plot

### LINKED ISSUE #19672: Spyder's `savefig` differs from plain terminal's
How does Spyder change `plt.savefig()` behavior? Comparison below. This creates major discrepancy between what Spyder generates and other applications (e.g. Jupyer, `sphinx-gallery`), while I need them to be same. Can Spyder be made to match terminal `python example.py`?

<hr>

```python
import matplotlib.pyplot as plt

plt.plot([1, 2])
plt.gcf().set_size_inches(6, 4)
plt.gcf().set_dpi(72)
plt.gcf().savefig("img.png")
```

**Command prompt**:

image

**Spyder**

image

with

image

### PR DESCRIPTION
**Pull request by jitseniesen**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Add two new options, for the font size and the position of the bottom edge, for inline plots in the IPython console. The reason for this was discussed in issue #19672. Send these options to the Spyder kernels, where they will be used in a partner PR (I hope I'm doing that right, I have never done this before).

When writing the code, I noticed that the (?) icons for the tooltips of some spinboxes (resolution, width, height) was missing, so I added these as well.

The Preferences page now looks like this:

gfxconf-new

For comparison, this is what it looks before the PR:

gfxconf-old

### Issue(s) Resolved

Fixes #19672
Depends on spyder-ide/spyder-kernels#477

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: Jitse Niesen

### CODE REVIEW

File: spyder/plugins/ipythonconsole/confpage.py [hunk 1]
```diff
@@ -171,6 +171,18 @@ def setup_page(self):
...
+                          _("Font size:")+"  ", " "+_("points"),
+                          'pylab/inline/fontsize', min_=5, max_=48, step=1.0,
+                          tip=_("Default is 10"))
+        bottom_spin = self.create_spinbox(
+                          _("Bottom edge:")+"  ",
+                          " "+_("of figure height"),
+                          'pylab/inline/bottom', min_=0, max_=0.3, step=0.01,
+                          tip=_("The position of the bottom edge of the "
+                                "subplots,\nas a fraction of the figure "
+                                "height.\nThe default is 0.11."))
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/confpage.py:184]:
```suggestion
        fontsize_spin = self.create_spinbox(
            _("Font size:") + "  ",
            " " + _("points"),
            'pylab/inline/fontsize',
            min_= 5,
            max_=48,
            step=1.0,
            tip=_("Default is 10")
        )
        bottom_spin = self.create_spinbox(
            _("Bottom edge:") + "  ",
            " " + _("of figure height"),
            'pylab/inline/bottom',
            min_=0,
            max_=0.3,
            step=0.01,
            tip=_("The position of the bottom edge of the subplots,\nas a "
                  "fraction of the figure height.\nThe default is 0.11.")
        )
```

Improve code style.

File: spyder/plugins/ipythonconsole/widgets/shell.py [hunk 2]
```diff
@@ -576,6 +578,18 @@ def send_mpl_backend(self, option=None):
                 if height_o is not None:
                     matplotlib_conf[height_n] = height_o
 
+            # Font size
+            fontsize_o = float(self.get_conf(fontsize_n))
+            if fontsize_o is not None and (
+                    option is None or fontsize_n in option):
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/widgets/shell.py:584]:
```suggestion
            if (
                fontsize_o is not None
                and (option is None or fontsize_n in option)
            ):
```

Improve code style.

File: spyder/plugins/ipythonconsole/widgets/shell.py [hunk 3]
```diff
@@ -576,6 +578,18 @@ def send_mpl_backend(self, option=None):
...
+                matplotlib_conf[fontsize_n] = fontsize_o
+
+            # Bottom part
+            bottom_o = float(self.get_conf(bottom_n))
+            if bottom_o is not None and (
+                    option is None or bottom_n in option):
```
Comment (ccordoba12) [inline spyder/plugins/ipythonconsole/widgets/shell.py:590]:
```suggestion
            if (
                bottom_o is not None
                and (option is None or bottom_n in option)
            ):
```

Improve code style.

File: spyder/plugins/variableexplorer/widgets/namespacebrowser.py [hunk 4]
```diff
@@ -497,11 +504,23 @@ def plot_in_plots_plugin(self, data, funcname):
             bbox_inches = 'tight'
         else:
             bbox_inches = None
+        matplotlib_rc = {
```
Comment (ccordoba12) [inline spyder/plugins/variableexplorer/widgets/namespacebrowser.py:507]:
```suggestion

        matplotlib_rc = {
```

File: spyder/plugins/variableexplorer/widgets/tests/test_namespacebrowser.py [hunk 5]
```diff
@@ -231,6 +231,35 @@ def test_namespacebrowser_plot_with_mute_inline_plotting_true(
...
+        from matplotlib import rcParams
+        assert rcParams['font.size'] == 20.5
+        assert rcParams['figure.subplot.bottom'] == 0.314
+
+    namespacebrowser.set_conf('mute_inline_plotting', True, section='plots')
+    namespacebrowser.plots_plugin_enabled = True
+    namespacebrowser.set_conf(
+        'pylab/inline/fontsize', 20.5, section='ipython_console')
+    namespacebrowser.set_conf(
+        'pylab/inline/bottom', 0.314, section='ipython_console')
```
Comment (ccordoba12) [inline spyder/plugins/variableexplorer/widgets/tests/test_namespacebrowser.py:249]:
```suggestion
        'pylab/inline/fontsize', 20.5, section='ipython_console'
    )
    namespacebrowser.set_conf(
        'pylab/inline/bottom', 0.314, section='ipython_console'
    )
```

**Review by ccordoba12 — COMMENTED** [review]
Thanks @jitseniesen for your work on this! Small small code style suggestions for you, the rest looks good to me.

**Comment by jitseniesen** [comment]
> Small small code style suggestions for you, the rest looks good to me.
> 
> Also, please resync our subrepo now that I merged your PR there.

Done, this should be good to go now if the tests pass.

**Review by ccordoba12 — APPROVED** [review]
Thanks @jitseniesen!

**Pull request merged**